### ISSUE TITLE
Internal Failure in `getPreloadTags` when running in production mode

### ISSUE THREAD

**Issue by chrisvariety** [brillout/vite-plugin-ssr#51]
Having trouble putting together a minimal reproduction for this, but the stack is below.

```
Error: [vite-plugin-ssr@0.1.0-beta.35][Internal Failure] You stumbled upon a bug in `vite-plugin-ssr`'s source code (an internal assertion failed). This should definitely not be happening, and you should create a new issue at https://github.com/brillout/vite-plugin-ssr/issues/new that includes this error stack (the error stack is enough to debug this). Or reach out on Discord. A fix will be written promptly.
    at /Users/chrismcc/workspace/ss/node_modules/vite-plugin-ssr/dist/getPreloadTags.node.js:33:22
    at Array.forEach (<anonymous>)
    at Object.getPreloadTags (/Users/chrismcc/workspace/ss/node_modules/vite-plugin-ssr/dist/getPreloadTags.node.js:23:22)
    at renderHtmlDocument (/Users/chrismcc/workspace/ss/node_modules/vite-plugin-ssr/dist/renderPage.node.js:222:53)
    at async renderPageId (/Users/chrismcc/workspace/ss/node_modules/vite-plugin-ssr/dist/renderPage.node.js:116:30)
    at async renderPage (/Users/chrismcc/workspace/ss/node_modules/vite-plugin-ssr/dist/renderPage.node.js:96:24)
    at async Object.<anonymous> (/Users/chrismcc/workspace/ss/server.js:1713:20)
```

Vite v2.2.3

It's trying to look up the manifest entry for `pages/all.page.tsx`, which doesn't exist in client or server manifest for some reason, even though the page does + works great in dev mode. It's meant to be a SSR page, I don't have any static or prerendered pages. It's also a catchall, the route is defined as:

```
export default function () {
  return {
    match: -2
  };
}
```

<details>
  <summary>Server Manifest</summary>
  
```js
{
  'node_modules/vite-plugin-ssr/page-files/pageFiles.node.ts': {
    file: 'pageFiles.node.js',
    src: 'node_modules/vite-plugin-ssr/page-files/pageFiles.node.ts',
    isEntry: true,
    dynamicImports: [
      'pages/_error.page.tsx',
      '_all.page.97a49683.js',
      'pages/onboard.page.tsx',
      'pages/session.page.tsx',
      'pages/shops.page.tsx',
      'pages/_default/_default.page.client.tsx',
      'pages/all.page.server.ts',
      'pages/onboard.page.server.tsx',
      'pages/session.page.server.tsx',
      'pages/shops.page.server.tsx',
      'pages/_default/_default.page.server.tsx',
      'pages/all.page.route.js',
      'pages/onboard.page.route.js',
      'pages/session.page.route.js',
      'pages/shops.page.route.js'
    ]
  },
  'pages/_error.page.tsx': {
    file: 'assets/_error.page.797e0037.js',
    src: 'pages/_error.page.tsx',
    isDynamicEntry: true,
    imports: [ '_TwoPanelLayout.5aed1da0.js' ]
  },
  '_TwoPanelLayout.5aed1da0.js': {
    file: 'assets/TwoPanelLayout.5aed1da0.js',
    assets: [ 'assets/pattern2x.bb73d710.png' ]
  },
  '_all.page.97a49683.js': {
    file: 'assets/all.page.97a49683.js',
    isDynamicEntry: true,
    imports: [
      '_index.4f9d37b7.js',
      '_use-graphql.4e21b4ca.js',
      '_types.62a059e8.js',
      'pages/_error.page.tsx'
    ],
    dynamicImports: [ 'components/PageEditor.tsx' ]
  },
  '_types.62a059e8.js': { file: 'assets/types.62a059e8.js' },
  '_index.4f9d37b7.js': { file: 'assets/index.4f9d37b7.js' },
  '_use-graphql.4e21b4ca.js': { file: 'assets/use-graphql.4e21b4ca.js' },
  'pages/onboard.page.tsx': {
    file: 'assets/onboard.page.b445335a.js',
    src: 'pages/onboard.page.tsx',
    isDynamicEntry: true,
    imports: [
      '_TwoPanelLayout.5aed1da0.js',
      '_BrandButton.36c3ecbe.js',
      '_graphql.e071fc6a.js',
      '_use-graphql.4e21b4ca.js',
      '_logo.65f94883.js',
      '_enums.f99be063.js',
      '_types.62a059e8.js'
    ],
    assets: [
      'assets/airtablekey900k.534aaf1b.mp4',
      'assets/airtablekey900k.5770f3fb.webm',
      'assets/airtableshare700k.bc2dac47.mp4',
      'assets/airtableshare700k.2685e756.webm',
      'assets/googleoutput700k.cbf09c54.mp4',
      'assets/googleoutput700k.0d8460f6.webm'
    ]
  },
  '_BrandButton.36c3ecbe.js': { file: 'assets/BrandButton.36c3ecbe.js' },
  '_graphql.e071fc6a.js': { file: 'assets/graphql.e071fc6a.js' },
  '_logo.65f94883.js': {
    file: 'assets/logo.65f94883.js',
    assets: [ 'assets/logo.85f87099.svg' ]
  },
  '_enums.f99be063.js': { file: 'assets/enums.f99be063.js' },
  'pages/session.page.tsx': {
    file: 'assets/session.page.32c80824.js',
    src: 'pages/session.page.tsx',
    isDynamicEntry: true,
    imports: [
      '_TwoPanelLayout.5aed1da0.js',
      '_graphql.e071fc6a.js',
      '_use-graphql.4e21b4ca.js',
      '_logo.65f94883.js'
    ]
  },
  'pages/shops.page.tsx': {
    file: 'assets/shops.page.9f3e6ba7.js',
    src: 'pages/shops.page.tsx',
    isDynamicEntry: true,
    imports: [ '_use-graphql.4e21b4ca.js', '_logo.65f94883.js' ]
  },
  'pages/_default/_default.page.client.tsx': {
    file: 'assets/_default.page.client.b5770ade.js',
    src: 'pages/_default/_default.page.client.tsx',
    isDynamicEntry: true
  },
  'pages/all.page.server.ts': {
    file: 'assets/all.page.server.e439a3de.js',
    src: 'pages/all.page.server.ts',
    isDynamicEntry: true,
    imports: [
      '_types.62a059e8.js',
      '_widget-fragment.15763a42.js',
      '_shop.699429dc.js'
    ]
  },
  '_widget-fragment.15763a42.js': { file: 'assets/widget-fragment.15763a42.js' },
  '_shop.699429dc.js': { file: 'assets/shop.699429dc.js' },
  'pages/onboard.page.server.tsx': {
    file: 'assets/onboard.page.server.fcbbdb8b.js',
    src: 'pages/onboard.page.server.tsx',
    isDynamicEntry: true,
    imports: [ '_enums.f99be063.js' ]
  },
  'pages/session.page.server.tsx': {
    file: 'assets/session.page.server.e7258d58.js',
    src: 'pages/session.page.server.tsx',
    isDynamicEntry: true
  },
  'pages/shops.page.server.tsx': {
    file: 'assets/shops.page.server.6f2e1061.js',
    src: 'pages/shops.page.server.tsx',
    isDynamicEntry: true,
    imports: [ '_shop.699429dc.js' ]
  },
  'pages/_default/_default.page.server.tsx': {
    file: 'assets/_default.page.server.09a6d56f.js',
    src: 'pages/_default/_default.page.server.tsx',
    isDynamicEntry: true,
    imports: [
      '_logo.65f94883.js',
      '_BrandButton.36c3ecbe.js',
      '_index.4f9d37b7.js'
    ],
    css: [ 'assets/_default.page.server.704ed2fd.css' ]
  },
  'pages/all.page.route.js': {
    file: 'assets/all.page.route.dc102e33.js',
    src: 'pages/all.page.route.js',
    isDynamicEntry: true
  },
  'pages/onboard.page.route.js': {
    file: 'assets/onboard.page.route.287b457d.js',
    src: 'pages/onboard.page.route.js',
    isDynamicEntry: true
  },
  'pages/session.page.route.js': {
    file: 'assets/session.page.route.cabbba58.js',
    src: 'pages/session.page.route.js',
    isDynamicEntry: true
  },
  'pages/shops.page.route.js': {
    file: 'assets/shops.page.route.e0261f54.js',
    src: 'pages/shops.page.route.js',
    isDynamicEntry: true
  },
  'components/PageEditor.tsx': {
    file: 'assets/PageEditor.9b8afbb3.js',
    src: 'components/PageEditor.tsx',
    isDynamicEntry: true,
    imports: [
      '_all.page.97a49683.js',
      '_use-graphql.4e21b4ca.js',
      '_types.62a059e8.js',
      '_widget-fragment.15763a42.js',
      '_index.4f9d37b7.js',
      'pages/_error.page.tsx',
      '_TwoPanelLayout.5aed1da0.js'
    ]
  }
}
```
</details>

<details>
  <summary>Client Manifest</summary>
  
```js
{
  'pages/_default/_default.page.client.tsx': {
    file: 'assets/pages/_default/_default.page.client.tsx.5bd00183.js',
    src: 'pages/_default/_default.page.client.tsx',
    isEntry: true,
    isDynamicEntry: true,
    imports: [ '_vendor.2ab1f7ad.js' ],
    css: [ 'assets/pages/_default/_default.page.client.tsx.d82a5d3f.css' ]
  },
  '_vendor.2ab1f7ad.js': {
    file: 'assets/vendor.2ab1f7ad.js',
    dynamicImports: [
      'pages/_error.page.tsx',
      '_all.page.15209e07.js',
      'pages/onboard.page.tsx',
      'pages/session.page.tsx',
      'pages/shops.page.tsx',
      'pages/_default/_default.page.client.tsx',
      'pages/all.page.route.js',
      'pages/onboard.page.route.js',
      'pages/session.page.route.js',
      'pages/shops.page.route.js'
    ]
  },
  'pages/_error.page.tsx': {
    file: 'assets/_error.page.a5a13a3d.js',
    src: 'pages/_error.page.tsx',
    isDynamicEntry: true,
    imports: [ '_TwoPanelLayout.2282149a.js', '_vendor.2ab1f7ad.js' ]
  },
  '_TwoPanelLayout.2282149a.js': {
    file: 'assets/TwoPanelLayout.2282149a.js',
    imports: [ '_vendor.2ab1f7ad.js' ],
    assets: [ 'assets/pattern2x.bb73d710.png' ]
  },
  '_all.page.15209e07.js': {
    file: 'assets/all.page.15209e07.js',
    isDynamicEntry: true,
    imports: [
      '_vendor.2ab1f7ad.js',
      'pages/_default/_default.page.client.tsx',
      '_use-graphql.968707a4.js',
      '_types.b4659228.js',
      'pages/_error.page.tsx'
    ],
    dynamicImports: [ 'components/PageEditor.tsx' ]
  },
  '_types.b4659228.js': { file: 'assets/types.b4659228.js' },
  '_use-graphql.968707a4.js': {
    file: 'assets/use-graphql.968707a4.js',
    imports: [ '_vendor.2ab1f7ad.js' ]
  },
  'pages/onboard.page.tsx': {
    file: 'assets/onboard.page.a6f81a5a.js',
    src: 'pages/onboard.page.tsx',
    isDynamicEntry: true,
    imports: [
      '_TwoPanelLayout.2282149a.js',
      'pages/_default/_default.page.client.tsx',
      '_graphql.4e80b5a8.js',
      '_vendor.2ab1f7ad.js',
      '_use-graphql.968707a4.js',
      '_logo.31f1c087.js',
      '_types.b4659228.js'
    ],
    assets: [
      'assets/airtablekey900k.534aaf1b.mp4',
      'assets/airtablekey900k.5770f3fb.webm',
      'assets/airtableshare700k.bc2dac47.mp4',
      'assets/airtableshare700k.2685e756.webm',
      'assets/googleoutput700k.cbf09c54.mp4',
      'assets/googleoutput700k.0d8460f6.webm'
    ]
  },
  '_graphql.4e80b5a8.js': {
    file: 'assets/graphql.4e80b5a8.js',
    imports: [ '_vendor.2ab1f7ad.js' ]
  },
  '_logo.31f1c087.js': {
    file: 'assets/logo.31f1c087.js',
    imports: [ '_vendor.2ab1f7ad.js' ]
  },
  'pages/session.page.tsx': {
    file: 'assets/session.page.205b7aa6.js',
    src: 'pages/session.page.tsx',
    isDynamicEntry: true,
    imports: [
      '_TwoPanelLayout.2282149a.js',
      '_graphql.4e80b5a8.js',
      '_use-graphql.968707a4.js',
      '_logo.31f1c087.js',
      '_vendor.2ab1f7ad.js',
      '_useEffectOnce.9167af87.js'
    ]
  },
  '_useEffectOnce.9167af87.js': {
    file: 'assets/useEffectOnce.9167af87.js',
    imports: [ '_vendor.2ab1f7ad.js' ]
  },
  'pages/shops.page.tsx': {
    file: 'assets/shops.page.9ce44a2b.js',
    src: 'pages/shops.page.tsx',
    isDynamicEntry: true,
    imports: [
      '_use-graphql.968707a4.js',
      '_logo.31f1c087.js',
      '_vendor.2ab1f7ad.js'
    ]
  },
  'pages/all.page.route.js': {
    file: 'assets/all.page.route.56b20769.js',
    src: 'pages/all.page.route.js',
    isDynamicEntry: true
  },
  'pages/onboard.page.route.js': {
    file: 'assets/onboard.page.route.d1d214e5.js',
    src: 'pages/onboard.page.route.js',
    isDynamicEntry: true
  },
  'pages/session.page.route.js': {
    file: 'assets/session.page.route.5daa49cd.js',
    src: 'pages/session.page.route.js',
    isDynamicEntry: true
  },
  'pages/shops.page.route.js': {
    file: 'assets/shops.page.route.6973bf40.js',
    src: 'pages/shops.page.route.js',
    isDynamicEntry: true
  },
  'components/PageEditor.tsx': {
    file: 'assets/PageEditor.df7c5aa1.js',
    src: 'components/PageEditor.tsx',
    isDynamicEntry: true,
    imports: [
      '_all.page.15209e07.js',
      '_vendor.2ab1f7ad.js',
      '_use-graphql.968707a4.js',
      '_useEffectOnce.9167af87.js',
      '_types.b4659228.js',
      'pages/_default/_default.page.client.tsx',
      'pages/_error.page.tsx',
      '_TwoPanelLayout.2282149a.js'
    ]
  }
}
```
</details>

Let me know if you need some additional info to hunt this one down...

**Comment by brillout** [comment]
This is indeed quite unexpected. (Your `*.page.*` files are loaded with Vite's `import.meta.glob()` so there is no reason why `pages/all.page.tsx` would be treated differently here.)

I'll need a reproduction for this one. If you remove all your others pages and just publish the problematic `pages/all.page.tsx` that would suffice.

**Comment by brillout** [comment]
Does your `components/PageEditor.tsx` import `pages/all.page.tsx`? You can try to extract what `PageEditor.tsx` needs in another non-`.page.` file, so that `pages/all.page.tsx` is not imported by anyone other than `vite-plugin-ssr`.

**Comment by chrisvariety** [comment]
@brillout Here's a reproduction: https://github.com/chrisvariety/vite-plugin-ssr/commit/69637ba520c3125eb8088864041320642503e738

Just run `NODE_ENV=production npm run prod` and then hit the index route. I can confirm removing the `lazy` resolves the issue, not sure why it's this specific setup beyond that.

**Comment by brillout** [comment]
Fix released in `0.1.0-beta.36`.

**Comment by brillout** [comment]
Reproduction with `0.4.0`: https://github.com/brillout/vps_repro_51.
